src/pages/Graph/SummaryPanelTraceDetails.tsx

```diff
@@ -155,42 +147,33 @@ class SummaryPanelTraceDetails extends React.Component<Props, State> {
               </div>
             )}
           </div>
-          {spans && (
+          {spans.length > 0 && (
             <div className={pStyle}>
               <div>
-                <strong>Spans for node: </strong>
-                {nodeName}
-              </div>
-              <div>
-                <Button
-                  className={navButtonStyle}
-                  variant={ButtonVariant.plain}
-                  isDisabled={this.state.selectedSpan === 0}
-                  onClick={_ => {
-                    if (this.state.selectedSpan > 0) {
-                      this.setState({ selectedSpan: this.state.selectedSpan - 1 });
-                    }
-                  }}
-                >
-                  <AngleLeftIcon />
-                </Button>
-                <span className={navSpanStyle}>{this.state.selectedSpan + 1 + ' of ' + spans.length}</span>
-                <Button
-                  variant={ButtonVariant.plain}
-                  isDisabled={this.state.selectedSpan >= spans.length - 1}
-                  onClick={_ => {
-                    if (this.state.selectedSpan < spans.length) {
-                      this.setState({ selectedSpan: this.state.selectedSpan + 1 });
-                    }
+                <strong>{pluralize(spans.length, 'span')}</strong> on this node
+                <SimplerSelect
+                  selections={currentSpan?.operationName}
+                  className={spanSelectStyle}
+                  onSelect={key => {
+                    this.setState({ selectedSpanID: key as string });
                   }}
                 >
-                  <AngleRightIcon />
-                </Button>
+                  {spans.map(s => {
+                    return (
+                      <SelectOption
+                        key={s.spanID}
+                        value={s.spanID}
+                        onClick={() => this.setState({ selectedSpanID: s.spanID })}
```

Comment:
Isn't this a duplicate of `onSelect` event?

Reply:
looks like it is, good catch!

Reply:
fixed